subsys/debug/tracing/CMakeLists.txt

```diff
@@ -1,27 +1,37 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(CONFIG_SEGGER_SYSTEMVIEW OR CONFIG_TRACING_CPU_STATS)
-  zephyr_library()
 
-  zephyr_library_sources_ifdef(
+  zephyr_sources_ifdef(
     CONFIG_SEGGER_SYSTEMVIEW
     sysview_config.c
     sysview.c
     )
 
-  zephyr_library_sources_ifdef(
+  zephyr_sources_ifdef(
     CONFIG_TRACING_CPU_STATS
     cpu_stats.c
     )
-
-  if(CONFIG_TRACING)
-    zephyr_library_include_directories(
-      ${ZEPHYR_BASE}/kernel/include
-      ${ZEPHYR_BASE}/arch/${ARCH}/include
-      )
-  endif()
 endif()
 
+zephyr_sources_ifdef(
+  CONFIG_TRACING
+  tracing_core.c
+  tracing_format.c
+  tracing_packet.c
```

Comment:
These sources should only be included under `CONFIG_TRACING_CTF_BOTTOM_GENERIC`.
E.g. Oticon has a proprietary CTF bottom (so CONFIG_TRACING is on) but we do not need this new buffered GENERIC bottom.

Reply:
Still unresolved. Please specialize from CONFIG_TRACING to something that actually depends on this, e.g. CONFIG_TRACING_GENERIC (or CONFIG_TRACING_BUFFERED, preferrable IMHO).